src/overlays/actors/ovl_Bg_Ddan_Jd/z_bg_ddan_jd.c

```diff
@@ -31,17 +33,157 @@ const ActorInit Bg_Ddan_Jd_InitVars = {
     (ActorFunc)BgDdanJd_Update,
     (ActorFunc)BgDdanJd_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ddan_Jd/BgDdanJd_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ddan_Jd/BgDdanJd_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+#define STATE_GO_BOTTOM 0
+#define STATE_GO_MIDDLE_FROM_BOTTOM 1
+#define STATE_GO_MIDDLE_FROM_TOP 2
+#define STATE_GO_TOP 3
+
+#define MOVE_HEIGHT_MIDDLE 140.0f
+#define MOVE_HEIGHT_TOP 700.0f
+
+#define IDLE_FRAMES 100
+
+// since stepSpeed also determines whether the platform is the shortcut to the top of the dungeon, these must be
+// different values to work correctly
+#define DEFAULT_STEP_SPEED 1
+#define SHORTCUT_STEP_SPEED 5
```

Comment:
not sold on calling it "shortcut" since it's just a gameplay intent rather than what the actor actually does (rise higher, two levels)

Reply:
i feel like gameplay intent is fine because the actor is a set piece designed to be used specifically in Dodongo's Cavern